blockmanager.go

```diff
@@ -484,6 +491,14 @@ func (b *blockManager) startSync() {
 		bmgrLog.Infof("Syncing to block height %d from peer %v",
 			bestPeer.LastBlock(), bestPeer.Addr())
 
+		// The chain is not synced whenever the current best height is less than
+		// the height to sync to.
+		b.isCurrentMtx.Lock()
+		if best.Height < bestPeer.LastBlock() {
+			b.isCurrent = false
+		}
+		b.isCurrentMtx.Unlock()
```

Comment:
Small nit. Every other instance accessing `isCurrent` does the lock inside the corresponding `if` vs this one where the lock is acquired before. 

There doesn't seem to be a reason for that, since both `best` and `bestPeer` are local and safe for concurrent access.

Reply:
Agreed.  Updated.